src/binheap.py

```diff
@@ -7,39 +7,47 @@ class BinHeap(object):
 
     def __init__(self, init=[]):
         """Create heap empty or with given values."""
-        self._container = [None]
+        self._container = []
         for element in init:
             self.push(element)
 
     def _sort_down(self, index):
         # Maintain heap property for children of given node(index).
+        index += 1
         left, right = 2 * index, 2 * index + 1
         largest = index
-        length = len(self._container) - 1
-
-        if left <= length and self._container[left] > self._container[largest]:
+        length = len(self._container)
+        if left <= length and self._container[left - 1] > self._container[largest - 1]:
             largest = left
-        if right <= length and self._container[right] > self._container[largest]:
+        if right <= length and self._container[right - 1] > self._container[largest - 1]:
             largest = right
         if largest != index:
-            this_node_value = self._container[index]
-            self._container[index] = self._container[largest]
-            self._container[largest] = this_node_value
-            self._sort_down(largest)
+            this_node_value = self._container[index - 1]
+            self._container[index - 1] = self._container[largest - 1]
+            self._container[largest - 1] = this_node_value
+            self._sort_down(largest - 1)
 
     def pop(self):
         """Remove root of heap and ensures heap is maintained."""
-        self._container[1] = self._container.pop()
-        self._sort_down(1)
+        if len(self._container) == 1:
+            self._container.pop()
+
+        elif len(self._container) > 1:
+            self._container[0] = self._container.pop()
+            self._sort_down(0)
+
+        else:
+            raise IndexError("The list is empty")
+
 
     def push(self, value):
         """Add given value to heap and maintain heap property."""
         self._container.append(value)
-        child = len(self._container) - 1
+        child = len(self._container)
         parent = child // 2
-        while (child > 1 and self._container[child] > self._container[parent]):
-            smaller_parent = self._container[parent]
-            self._container[parent] = self._container[child]
-            self._container[child] = smaller_parent
+        while (child > 1 and self._container[child - 1] > self._container[parent -1]):
+            smaller_parent = self._container[parent -1]
+            self._container[parent-1] = self._container[child-1]
+            self._container[child-1] = smaller_parent
```

Comment:
Be consistent with your formatting. You have `[child - 1]`, `[parent -1]`, and `[parent-1]` --- 3 separate whitespace standards. Pick one and stick to it. For this class we are adhering to the flake8 standard E225 regarding whitespace around operators. This quote from the docs is what you should take away from this class:

> A style guide is about consistency. Consistency with this style guide is important. Consistency within a project is more important. Consistency within one module or function is most important.